core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -1859,16 +1843,21 @@ public void run() {
             }
             Attributes attributes = attrBuilder.build();
 
-            boolean addressesAccepted = helper.lb.tryAcceptResolvedAddresses(
+            lastAddressesAccepted = helper.lb.tryAcceptResolvedAddresses(
                 ResolvedAddresses.newBuilder()
                     .setAddresses(servers)
                     .setAttributes(attributes)
                     .setLoadBalancingPolicyConfig(effectiveServiceConfig.getLoadBalancingConfig())
                     .build());
+          } else {
+            lastAddressesAccepted = false;
```

Comment:
Getting here means the NR has been shut down. I think saying `true` here is a bit easier to deal with, but `false` does make sense. In either case, we need RetryingNameResolver to _not_ schedule a backoff. Right now, `RetryingNameResolver.shutdown()` calls `retryScheduler.reset()`, but calling `retryScheduler.schedule()` after `reset()` will schedule a new backoff.

The easiest thing may be to just have `RetryingNameResolver` have a `bool shutdown` field (synchronized using SynchronizationContext) that is checked in `resolutionAttempted()` and `onError()`.

Reply:
Thinking about this some more, it seems that if the NR is shut down we are already in a "game over" state and it's not really necessary to call the listener to say the addresses were not accepted - there won't be any retries anyway.

I'm now just passing the listener what the LB reports back.